instrumentation/redis/test/opentelemetry/instrumentation/redis/patches/client_test.rb

```diff
@@ -297,7 +338,7 @@ def redis_with_auth(redis_options = {})
         _(exporter.finished_spans.size).must_equal 3
 
         set_span = exporter.finished_spans[0]
-        _(set_span.name).must_equal 'AUTH'
+        _(set_span.name).must_equal(redis_gte_5? ? 'PIPELINED' : 'AUTH')
```

Comment:
We may want to look into splitting these tests up. I know it will lead to duplication but conditional logic in tests that go beyond using `skip` become difficult to reason about IMO.

Reply:
Do you mean to individually duplicate the tests that should be forked, or duplicate the file entirely for v5?